contract/bridge/contracts/VaultReward.sol

```diff
@@ -0,0 +1,192 @@
+
+  function extendVaultLockPeriod(address _vaultId, uint256 _lockPeriod) external vaultExist(_vaultId) {
+    require(_vaultId == msg.sender, "Invalid vaultId");
+
+    // check if the lockPeriod is valid
+    require(_lockPeriod == 3 || _lockPeriod == 6 || _lockPeriod == 12, "Lock period should be one of 3, 6, 12");
+
+    // update vault accClaimableRewards
+    updateVaultAccClaimableRewards(_vaultId);
+
+    // get vault
+    LockedVault storage vault = lockedVaults[_vaultId];
+
+    // get the current vault lock period
+    uint256 oldLockPeriod = vault.lockPeriod;
+
+    // update the vault lock info
+    uint256 secPerDay = 60 * 60 * 24;
+    uint256 lockPeriodInSec = secPerDay.mul(30).mul(_lockPeriod);
+    if (vault.lockExpireAt < block.timestamp) {   // new or expired vault
+      vault.lockStartAt = block.timestamp;
+      vault.lockPeriod = _lockPeriod;
+      vault.lockExpireAt = block.timestamp.add(lockPeriodInSec);
+      vault.rewardClaimAt = block.timestamp;
+      vault.accClaimableRewards = 0;
+      vault.accRewardPerShare = 0;
+      vault.accRewardPerSharelUpdatedAt = block.timestamp;
+      vault.collateralDebt = 0;
+    } else {    // locked vault
+      vault.lockPeriod = vault.lockPeriod.add(_lockPeriod);
+      vault.lockExpireAt = vault.lockExpireAt.add(lockPeriodInSec);
+    }
+
+    emit ExtendVaultLockPeriod(_vaultId, oldLockPeriod, vault.lockPeriod);
+  }
+
+  function updateVaultAccClaimableRewards(address _vaultId) public {
+    if (block.timestamp <= lockedVaults[_vaultId].lockExpireAt) {
+      // get vault
+      LockedVault storage vault = lockedVaults[_vaultId];
+
+      // store the old accClaimableRewards
+      uint256 oldAccClaimableRewards = vault.accClaimableRewards;
+
+      // update the vault info
+      (uint256 claimableRewards, uint256 rewardClaimAt) = getClaimableRewards(_vaultId);
+      vault.accClaimableRewards = claimableRewards;
+      vault.rewardClaimAt = rewardClaimAt;
+
+      emit UpdateVaultAccClaimableRewards(_vaultId, oldAccClaimableRewards, vault.accClaimableRewards, rewardClaimAt);
+    }
+  }
+
+  function claimRewards(address payable _vaultId) external {
+    require(_vaultId == msg.sender, "Invalid vaultId");
+
+    // claiim rewards
+    (uint256 claimableRewards, uint256 rewardClaimAt) = _claimRewards(_vaultId, _vaultId);
+
+    emit ClaimRewards(_vaultId, claimableRewards, rewardClaimAt);
+  }
+
+  function claimRewardsAndLock(address payable _vaultId) external {
+    require(_vaultId == msg.sender, "Invalid vaultId");
+
+    // claim rewards
+    (uint256 claimableRewards, uint256 rewardClaimAt) = _claimRewards(_vaultId, address(this));
+
+    // lock collateral
+    IVaultRegistry(oneBtc).lockAdditionalCollateralFromVaultReward{value: claimableRewards}(_vaultId);
+
+    emit ClaimRewardsAndLock(_vaultId, claimableRewards, rewardClaimAt);
+  }
+
+  function _claimRewards(address _vaultId, address payable _to) internal returns(uint256, uint256) {
+    // get reward debt
```

Comment:
Need vault exist check?

Reply:
Yes, Need vault existence checking. It's checked on getClaimableRewards() function.